core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -127,9 +129,12 @@ public TieredBlockStore() {
 
     BlockMetadataEvictorView initManagerView = new BlockMetadataEvictorView(mMetaManager,
         Collections.<Long>emptySet(), Collections.<Long>emptySet());
-    mAllocator = Allocator.Factory.create(initManagerView);
-    if (mAllocator instanceof BlockStoreEventListener) {
-      registerBlockStoreEventListener((BlockStoreEventListener) mAllocator);
+
+
+//    mAllocator = Allocator.Factory.create(initManagerView);
+    mAllocationCoordinator = AllocationCoordinator.getInstance(initManagerView);
+    if (mAllocationCoordinator.getAllocator() instanceof BlockStoreEventListener) {
```

Comment:
Currently none of the `Allocator` implementations implements `BlockStoreEventListener` but for the sake of backward compatibility I'm keeping this.